src/bin/pg_autoctl/monitor.c

```diff
@@ -496,6 +548,25 @@ parseNodeState(void *ctx, PGresult *result)
 		++errors;
 	}
 
+	value = PQgetvalue(result, 0, 3);
+	if (sscanf(value, "%d", &context->assignedState->candidatePriority) != 1)
```

Comment:
We don't use `sscanf`, please see about using `strtol` and friends.

Reply:
We used sscanf in the same function, please see how we parse nodeId and groupId fields.
We should do the same for all if we want to switch to strtol family